request-handler/src/main/java/au/org/aodn/aws/wps/QueryStringParameterParser.java

```diff
@@ -1,18 +1,100 @@
 package au.org.aodn.aws.wps;
 
+import au.org.aodn.aws.wps.operation.DescribeProcessOperation;
+import au.org.aodn.aws.wps.operation.GetCapabilitiesOperation;
 import au.org.aodn.aws.wps.operation.Operation;
+import au.org.aodn.aws.wps.operation.OperationFactory;
+import net.opengis.ows._1.AcceptVersionsType;
+import net.opengis.ows._1.CodeType;
+import net.opengis.wps._1_0.DescribeProcess;
+import net.opengis.wps._1_0.GetCapabilities;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.Map;
+import java.util.StringTokenizer;
 
 public class QueryStringParameterParser implements RequestParser {
+
+    Logger LOGGER = LoggerFactory.getLogger(QueryStringParameterParser.class);
+
     private final Map<String, String> queryParameters;
 
+    private final String SERVICE_REQUEST_PARAMETER_NAME = "service";
+    private final String REQUEST_NAME_PARAMETER_NAME = "request";
+    private final String VERSION_REQUEST_PARAMETER_NAME = "version";
+    private final String IDENTIFIER_REQUEST_PARAMETER_NAME = "identifier";
+
+
     public QueryStringParameterParser(AwsApiRequest request) {
         this.queryParameters = request.getQueryStringParameters();
     }
 
     @Override
     public Operation getOperation() {
-        throw new UnsupportedOperationException("Not yet implemented");
+        if(queryParameters.get(REQUEST_NAME_PARAMETER_NAME) != null)
+        {
+
+            String request = queryParameters.get(REQUEST_NAME_PARAMETER_NAME);
+
+            LOGGER.info("Request name: " + request);
+
+            Operation operation = OperationFactory.getInstance(request);
+
+            LOGGER.info("Operation returned: " + operation.getClass().getName());
+
+            if (operation instanceof DescribeProcessOperation)
+            {
+                //  Set relevant parameters
+                DescribeProcessOperation describeOperation = (DescribeProcessOperation) operation;
+
+                DescribeProcess describeRequest = describeOperation.getRequest();
```

Comment:
Looks to me like you actually wanted to the OGC request above and then the operation (using the existing getOperation in OperationFactory)

Reply:
Will revisit.  This is working currently + will do for this iteration.